go-runtime/encoding/encoding_test.go

```diff
@@ -77,14 +87,19 @@ func TestUnmarshal(t *testing.T) {
 			String string
 			Unit   ftl.Unit
 		}{String: "something", Unit: ftl.Unit{}}},
+		// Whitespaces after each `:` and multiple fields to test handling of the
+		// two potential terminal delimiters: `}` and `,`
+		{name: "ComplexFormatting", input: `{"option": null, "bool": true}`, expected: struct {
+			Option ftl.Option[int]
+			Bool   bool
+		}{ftl.None[int](), true}},
+		{name: "EmptyPointer", input: "null", expected: emptyPtr},
```

Comment:
Was this the test case we missed that broke PFI?

Reply:
it was because their outermost type was a pointer (`*builtin.HttpRequest[...]`), and all of our tests wrap everything in a value struct so we weren't exercising that :(

Reply:
oop nvm, pointers are not supported! updating to explicitly error instead of accommodate this case, since we don't have pointers in the schema